File: java/plugins/wrappers/java/nsXPIDLPluginTagInfo2.java
/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
 * The contents of this file are subject to the Mozilla Public
 * License Version 1.1 (the "License"); you may not use this file
 * except in compliance with the License. You may obtain a copy of
 * the License at http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS
 * IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
 * implied. See the License for the specific language governing
 * rights and limitations under the License.
 *
 * The Original Code is mozilla.org code.
 *
 * The Initial Developer of the Original Code is Sun Microsystems,
 * Inc. Portions created by Sun are
 * Copyright (C) 1999 Sun Microsystems, Inc. All
 * Rights Reserved.
 *
 * Contributor(s):
 * Serge Pikalev <sep@sparc.spb.su>
 */

import org.mozilla.xpcom.*;
import org.mozilla.pluglet.*;
import org.mozilla.pluglet.mozilla.*;

public class nsXPIDLPluginTagInfo2 extends nsXPIDLPluginTagInfo implements PlugletTagInfo2 {

    public nsIXPIDLPluginTagInfo2 tagInfo;

    public nsXPIDLPluginTagInfo2( nsIXPIDLPluginTagInfo2 tagInfo ) {
        this.tagInfo = tagInfo;
    }

    public String getAlignment() {
        return tagInfo.getAlignment();
    }

    public String getParameter( String name ) {
        return tagInfo.getParameter( name );
    }

    public java.util.Properties getParameters() {
        // get atts
        int[] count = new int[1];
        String[][] names = new String[1][];
        String[][] values = new String[1][];
        tagInfo.getParameters( count, names, values );

        // create props
        java.util.Properties props = new java.util.Properties();
        for( int i = 0; i < count[0]; i++ ) {
            props.setProperty( names[0][i], values[0][i] );
        }

        // return props
        return props;
    }

    public int getBorderHorizSpace() {
        return tagInfo.getBorderHorizSpace();
    }

    public int getBorderVertSpace() {
        return tagInfo.getBorderVertSpace();
    }

    public String getDocumentBase() {
        return tagInfo.getDocumentBase();
    }

    public String getDocumentEncoding() {
        return tagInfo.getDocumentEncoding();
    }

    public String getTagText() {
        return tagInfo.getTagText();
    }

    public String getTagType() {
        return tagInfo.getTagType();
    }

    public int getUniqueID() {
        return tagInfo.getUniqueID();
    }

    public int getHeight() {
        return tagInfo.getHeight();
    }

    public int getWidth() {
        return tagInfo.getWidth();
    }

    public Object getDOMElement() {
        return null;
    }
}

File: java/plugins/wrappers/cpp/nsXPIDLPluginTagInfo2.cpp
/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
 * The contents of this file are subject to the Mozilla Public
 * License Version 1.1 (the "License"); you may not use this file
 * except in compliance with the License. You may obtain a copy of
 * the License at http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS
 * IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
 * implied. See the License for the specific language governing
 * rights and limitations under the License.
 *
 * The Original Code is mozilla.org code.
 *
 * The Initial Developer of the Original Code is Sun Microsystems,
 * Inc. Portions created by Sun are
 * Copyright (C) 1999 Sun Microsystems, Inc. All
 * Rights Reserved.
 *
 * Contributor(s):
 * Serge Pikalev <sep@sparc.spb.su>
 */

#include "nsXPIDLPluginTagInfo2.h"

NS_IMPL_ISUPPORTS1(nsXPIDLPluginTagInfo2, nsIXPIDLPluginTagInfo2)

nsXPIDLPluginTagInfo2::nsXPIDLPluginTagInfo2()
{
  NS_INIT_ISUPPORTS();
}

nsXPIDLPluginTagInfo2::nsXPIDLPluginTagInfo2( nsIPluginTagInfo2 *tagInfo )
{
  NS_INIT_ISUPPORTS();
  this->tagInfo = tagInfo;
  NS_ADDREF( tagInfo );
}

nsXPIDLPluginTagInfo2::~nsXPIDLPluginTagInfo2()
{
  NS_RELEASE( tagInfo );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetAlignment( char * *aAlignment )
{
    return tagInfo->GetAlignment( (const char **)aAlignment );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetParameter( const char *name, char **_retval )
{
    return tagInfo->GetParameter( name, (const char **)_retval );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetParameters( PRUint32 *count,
                                      char ***names,
                                      char ***values )
{
    return tagInfo->GetParameters( (PRUint16)*count,
                                   (const char*const*)*names,
                                   (const char*const*)*values );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetBorderHorizSpace( PRUint32 *aBorderHorizSpace )
{
    return tagInfo->GetBorderHorizSpace( aBorderHorizSpace );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetBorderVertSpace( PRUint32 *aBorderVertSpace )
{
    return tagInfo->GetBorderVertSpace( aBorderVertSpace );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetDocumentBase( char * *aDocumentBase )
{
    return tagInfo->GetDocumentBase( (const char **)aDocumentBase );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetDocumentEncoding( char * *aDocumentEncoding )
{
    return tagInfo->GetDocumentEncoding( (const char**)aDocumentEncoding );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetHeight( PRUint32 *aHeight )
{
    return tagInfo->GetHeight( aHeight );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetWidth( PRUint32 *aWidth )
{
    return tagInfo->GetWidth( aWidth );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetTagText( char * *aTagText )
{
    return tagInfo->GetTagText( (const char **)aTagText );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetTagType( char * *aTagType )
{
    return tagInfo->GetTagType( (nsPluginTagType *)aTagType );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetUniqueID( PRUint32 *aUniqueID )
{
    return tagInfo->GetUniqueID( aUniqueID );
}

// from TagInfo
NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetAttribute( const char *name, char **_retval )
{
    return nsXPIDLPluginTagInfo::GetAttribute( name, _retval );
}

NS_IMETHODIMP
nsXPIDLPluginTagInfo2::GetAttributes( PRUint32 *count,
                                      char ***names,
                                      char ***values )
{
    return nsXPIDLPluginTagInfo::GetAttributes( count, names, values );
}
